tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1181,9 +1181,23 @@ static Optional<AccessLevel> inferAccessSyntactically(const ValueDecl *D) {
   llvm_unreachable("Unhandled DeclContextKind in switch.");
 }
 
+/// Document structure is a purely syntactic request that shouldn't require name lookup
+/// or type-checking, so this is a best-effort computation.
+static bool inferIsSettableSyntactically(const AbstractStorageDecl *D) {
+  if (auto *VD = dyn_cast<VarDecl>(D)) {
+    if (VD->isLet())
+      return false;
```

Comment:
I think we might need to do something like:
```
if (VD->hasInitialValue())
  return true;
```
here to handle cases like the below, which doesn't have a parsed setter, but is still settable:
```
var x = 10 {
  didSet { print(oldValue) }
}
```

Reply:
`hasInitialValue()` seems like it could trigger request evaluator, I've made a change to also check for `WillSet/DidSet` accessors.